datalad_fuse/tests/test_fuse.py

```diff
@@ -0,0 +1,18 @@
+import subprocess
+
+from datalad.tests.utils import assert_in_results
+import pytest
+
+
+@pytest.mark.libfuse
+def test_fuse(data_files, tmp_path, url_dataset):
+    assert_in_results(
+        url_dataset.fusefs(str(tmp_path)),
+        action="fusefs",
+        type="dataset",
+        path=str(tmp_path),
+        status="ok",
+    )
+    for fname, blob in data_files.items():
+        assert (tmp_path / fname).read_bytes() == blob
+    subprocess.run(["fusermount", str(tmp_path)], check=True)
```

Comment:
shouldn't this have `-u` (unmount) option given?

Reply:
The `fusermount` manpage implies that it unmounts by default and doesn't indicate when the `-u` option would actually be useful.  I also can't test it out in a Docker container as FUSE depends on installing a kernel module.

Reply:
I will test it out shortly.
I guess it would be useful anyways to have some Linux VM on your OSX - this way you can be in control of anything you need in the test environment.

Reply:
this is what I see 
```
$> mount | grep /tmp/fuse
DataLadFUSE on /tmp/fuse type fuse (rw,nosuid,nodev,relatime,user_id=47521,group_id=47522)
1 17539.....................................:Wed 29 Sep 2021 05:05:24 PM EDT:.
lena:/tmp
$> fusermount /tmp/fuse  
fusermount: old style mounting not supported
1 17540 ->1.....................................:Wed 29 Sep 2021 05:05:26 PM EDT:.
lena:/tmp
$> fusermount --version
fusermount3 version: 3.10.3
$> fusermount -u /tmp/fuse
$>
```
so works for me with `-u` and doesn't work without.

Re docker/fuse OSX, have a look at https://stackoverflow.com/questions/48402218/fuse-inside-docker which e.g. has


I know I'm late to the party but re: /dev/fuse on OS X, Fuse for OS X works great and can be installed via brew: brew install osxfuse or osxfuse.github.io – 
Chaim Eliyah
 Jun 17 '19 at 3:06

Reply:
re running tests on github actions -- have a look e.g. at https://github.com/s3fs-fuse/s3fs-fuse/tree/master/.github/workflows how to set it up on various systems